controller_manager:
  ros__parameters:
    update_rate: 50  # Hz
    diagnostic_period: 1

    effort_controller:
      type: forward_command_controller/ForwardCommandController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    position_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

effort_controller:
  ros__parameters:
    joints:
      - J11
      - J12
      - J13
      - J21
      - J22
      - J23
      - J31
      - J32
      - J33
      - J41
      - J42
      - J43
      - J51
      - J52
      - J53
      - J61
      - J62
      - J63
    interface_name: effort

position_trajectory_controller:
  ros__parameters:
    joints:
      - J11
      - J12
      - J13
      - J21
      - J22
      - J23
      - J31
      - J32
      - J33
      - J41
      - J42
      - J43
      - J51
      - J52
      - J53
      - J61
      - J62
      - J63
    command_interfaces: [position]
    state_interfaces: [position]

imu:
  ros__parameters:
    port: /dev/ttyAMA4
    frame_id: imu_link
    frequency: 30.0
    self_manage: true
    use_magnetometer: true

    angular_velocity_stdev: 0.01
    linear_acceleration_stdev: 0.0015
    magnetic_field_stdev: 0.0000005
    orientation_stdev: 0.000001
    axis_map: 0x24
    axis_sign: 0x06
    # these calibration values need to be determined on a per-device basis.
    calibration/accelerometer_offset: [1280, -9473, -6913]
    calibration/magnetometer_offset: [-24832, -3072, 22274]
    calibration/gyroscope_offset: [-1, 0, 0]
    calibration/accelerometer_radius: -6141
    calibration/magnetometer_radius: -8000


robot_dynamics:
  ros__parameters:
    frequency: 20.0
    model_state_frequency: 10.0
    self_manage: true
    leg_support_distance: 0.2
    joint_states: joint_states
    odometry: /odom_raw
    joint_controller: position_trajectory_controller
    effort_controller: /effort_controller/commands

robot_control:
  ros__parameters:
    frequency: 40.0
    publish_state_frequency: 20.0
    preview_frequency: 15.0
    progress_frequency: 10.0
    self_manage: true
    joint_controller: position_trajectory_controller
    chain0: "[leg] base_link => r_sole"
    chain1: "[leg] base_link => l_sole"
    joint_names:
      - J11
      - J12
      - J13
      - J21
      - J22
      - J23
      - J31
      - J32
      - J33
      - J41
      - J42
      - J43
      - J51
      - J52
      - J53
      - J61
      - J62
      - J63


